### DISCUSSION TITLE
Return invalid value in message of custom validator class.

### DISCUSSION THREAD

**Discussion by SchillingPilot** [adonisjs/core#2452] [Help] [answered]
I've been looking for a way to return the value of the field that failed validation using a `Custom Validator Class`. With single unique fields, it is easy to instruct the user as the to bad value so they can correct accordingly. This is not the case with an array of values that could be one or more in length.

### CustomValidator.ts
```javascript
public schema = schema.create({
    emails: schema
      .array([rules.minLength(1)])
      .members(
        schema.object().members({ email: schema.string({}, [rules.email()]) })
      ),
})

public messages = {
    'emails.minLength': 'Please provide at least one valid email address.',
    'emails.*.email.email': 'Invalid email: {{ value }}',
  }
```
Unfortunately, this is what I get back when the validation fails:
```json
{
    "errors": [
        {
            "rule": "email",
            "field": "emails.0.email",
            "message": "Invalid email: undefined"
        }
    ]
}
```
I would like to push the actual input from the user into the message field, so given an input like such as
```json
"emails": [
        {
            "email": "NewUserEmail1email.com", // <--- The bad value.
            "type": "personal"
        },
        {
            "email": "NewUserEmail2@email.com",
            "type": "business"
        }
]
```
I would like to get the following output...
```json
{
    "errors": [
        {
            ...
            "message": "Invalid email: NewUserEmail1email.com"
        }
    ]
}
```
Is there simple a way to accomplish this? 
Thanks.

**Comment by thetutlage** [comment, marked as answer]
Nope, we don't pass the value to the custom messages. I am not sure if it is a one line change or needs big refactor. You can look into the codebase of validator and feel free to send a PR to add the value as well.

**Reply by SchillingPilot** [reply]
Understood. Thank you for your help!
